src/javaharness/java/arcs/api/Type.java

```diff
@@ -0,0 +1,38 @@
+package arcs.api;
+
+public class Type {
+    enum Tag {
+        UNKNOWN, ENTITY, COLLECTION;
+        public boolean isCollection() {
+            return this == COLLECTION;
+        }
+        public boolean isEntity() {
+            return this == ENTITY;
+        }
+        static Tag fromString(String tag) {
```

Comment:
Since the enum tag name is the same as the stringified name except for case, you can rewrite this as:

static Tag fromString(String tag) { try { return Tag.valueOf(tag.toUpperCase()); } catch(Exception e) { return UNKNOWN; } }

Reply:
done